### PR TITLE
ddl: fix a bug when we backfill index with nil

### PR DESCRIPTION
**Pull request by zimulala**

### CODE REVIEW

File: ddl/ddl_db_test.go [hunk 1]
```diff
@@ -454,6 +454,27 @@ func (s *testDBSuite) testAlterLock(c *C) {
...
+func (s *testDBSuite) TestAddMultiColumnsIndex(c *C) {
+	s.tk = testkit.NewTestKit(c, s.store)
+	s.tk.MustExec("use " + s.schemaName)
+
+	s.tk.MustExec("drop database if exists tidb;")
+	s.tk.MustExec("create database tidb;")
+	s.tk.MustExec("use tidb;")
+	s.tk.MustExec("create table tidb.test (a int auto_increment primary key, b int);")
+	s.tk.MustExec("insert tidb.test values (1, 1);")
+	s.tk.MustExec("begin;")
```
Comment (hanfei1991) [inline ddl/ddl_db_test.go]: This test is not easy to understand. Please try to simplify it. Some command like 'begin' or 'commit' is not needed,  `insert` or `insert into` appears everywhere.

**Comment by XuHuaiyu** [comment]
LGTM

**Pull request merged**